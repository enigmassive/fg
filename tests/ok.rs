use std::fs;

#[test]
fn test() {
    fs::read_dir("./tests/ok/").unwrap().for_each(|entry| {
        let entry = entry.unwrap();
        let name = entry.file_name().into_string().unwrap();
        let path = entry.path();

        let src = String::from_utf8(fs::read(path).unwrap()).unwrap();
        let mut dest = vec![];
        fg::compile(&src, &mut dest);
        let code = String::from_utf8(dest).unwrap();
        insta::assert_snapshot!(name, code);
    })
}
