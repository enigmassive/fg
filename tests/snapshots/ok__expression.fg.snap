---
source: tests/ok.rs
expression: code
---
package expression

func expressions() {
	var a = f(0)
	var b = 0
	b, a = h(f(1), (f(2)))
}

func f(a int) (b int) {
	return a
}

func g(a int, b int) (c int, d int) {
	c = b
	d = a
	return c, d
}

func h(a int, b int) (int, int) {
	var c, d = g(a, b)
	return c, d
}
